### PR TITLE
fix: [PIE-3509]: flowControl tab now has apply changes and discard button

### PR DESCRIPTION
**Pull request by satyam-harness**
##### Jira Links:
https://harness.atlassian.net/browse/PIE-3509

##### Screenshots:

before 
Screenshot 2022-04-20 at 7 12 37 PM

after
Screenshot 2022-04-20 at 7 12 47 PM

Screenshot 2022-04-20 at 7 12 53 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger jest

**Review by samarth-sharma31 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx [hunk 1]
```diff
@@ -46,4 +65,105 @@ describe('FlowControl tests', () => {
...
+          <RightDrawer />
+        </TestWrapper>
+      </PipelineContext.Provider>
+    )
+
+    const addBarrierButton = await findByText('pipeline.barriers.addBarrier')
+    expect(addBarrierButton).toBeTruthy()
+    act(() => {
+      fireEvent.click(addBarrierButton)
+    })
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx:90]: After click write some assertion which verifies that addition happened successfully.

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx [hunk 2]
```diff
@@ -46,4 +65,105 @@ describe('FlowControl tests', () => {
...
+          <RightDrawer />
+        </TestWrapper>
+      </PipelineContext.Provider>
+    )
+
+    const applyChangeButton = await findByText('applyChanges')
+    expect(applyChangeButton).toBeTruthy()
+    act(() => {
+      fireEvent.click(applyChangeButton)
+    })
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx:144]: Write assertion regarding updatePipeline function call, it should be called with correct arguments.

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx:99]: addressed

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx]: addressed

**Review by vkbansal — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx [hunk 3]
```diff
@@ -46,4 +65,92 @@ describe('FlowControl tests', () => {
...
+    await waitFor(() => fireEvent.click(getByText('pipeline.barriers.syncBarriers')))
+
+    expect(container.querySelectorAll('[data-icon="main-trash"]').length).toBe(1)
+
+    const applyChangeButton = await findByText('applyChanges')
+    expect(applyChangeButton).toBeTruthy()
+    await act(() => {
+      fireEvent.click(applyChangeButton)
+    })
+    await waitFor(() => expect(pipelineContextMock.updatePipeline).toHaveBeenCalled())
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx:100]: Write assertion for to check if correct arguments are being passed to updatePipeline call.

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx (hunk 2 above)
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx]: Looks like this is removed and merged with first test case. However, first test case is not addressing this suggestion.

File: src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx (hunk 3 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineStudio/FlowControl/__tests__/FlowControl.test.tsx:100]: tried that but things were working fine on UI, but in test case state was not getting updated somehow so arguments were not proper, will take care of it in next PR

**Pull request merged**